modules/tensor_mechanics/include/actions/TensorMechanicsAction.h

```diff
@@ -66,6 +67,9 @@ class TensorMechanicsAction : public TensorMechanicsActionBase
   /// set generated from the combined block restrictions of all TensorMechanics/Master action blocks
   std::set<SubdomainID> _subdomain_id_union;
 
+  /// this will be true for only one single TensorMechanics (or derived) action
+  bool _is_primary_action;
```

Comment:
Should deriving classes overwrite this variable, or is it handled automagically somehow? I see only the initialization to false on ln 103 of the C file

Reply:
no, this should not be touched. I might make this member private